lib/start-command.js

```diff
@@ -0,0 +1,132 @@
+'use strict';
+
+var debug = require('./debug')('start-command');
+var fs = require('fs');
+var path = require('path');
+
+module.exports = exports = resolveArgs;
+
+// for tests
+exports.fromPath = fromPath;
+exports.resolveFile = resolveFile;
+exports.resolveDir = resolveDir;
+exports.fromStart = fromStart;
+
+function resolveArgs(cwd, argv) {
+  var script = argv[2] || '.';
+  return fromPath(cwd, script);
+}
+
+function fromPath(cwd, script) {
+  var app = {
+    cwd: cwd,
+    path: script,
+    error: null,
+  };
+
+  try {
+    var stat = fs.statSync(path.resolve(app.cwd, app.path));
+  } catch (er) {
+    try {
+      app.path = require.resolve(path.resolve(app.cwd, app.path));
+      stat = fs.statSync(app.path);
+    } catch (er) {
+      app.error = er;
+      return app;
+    }
+  }
+
+  // We want to be in the directory of the file we are running, so we can pick
+  // up configuration stored in it's working directory.
+
+  if (stat.isFile()) {
+    // The app is a file, we'll run it from its module's root directory.
+    return resolveFile(cwd, script);
+  } else if (stat.isDirectory()) {
+    // The app is a directory (such as '.'), we'll look for what to run from it.
+    try {
+      require.resolve(path.resolve(cwd, script));
```

Comment:
I commented more below, but I'm pretty sure that as-written, the resolution is:
- if there is no package.main or index.js
  - try to parse package.start
- else
  - use first of server.js, app.js, package.main, index.js

which means that if you don't have a package.main or index.js or package.scripts.start, you server.js and app.js will never be used either?